### PR TITLE
RQLY-1206: fix dropdown selections in rule editors

### PR DESCRIPTION
**Pull request by RuntimeTerror10**
Closes issue: 

## ✅ Checklist:

- [x] Make sure linting and unit tests pass.
- [x] No install/build warnings introduced.
- [x] Verified UI in browser.
- [x] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/RequestSourceRow/index.js [hunk 1]
```diff
@@ -26,108 +26,101 @@ const RequestSourceRow = ({
...
+        onClick={(event) =>
+          modifyPairAtGivenPath(
+            event,
+            pairIndex,
+            APP_CONSTANTS.PATH_FROM_PAIR.RULE_KEYS,
+            GLOBAL_CONSTANTS.RULE_KEYS.URL
+          )
+        }
+      >
+        URL
```
Comment (rohanmathur91) [inline app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/RequestSourceRow/index.js:40]: We should refactor this to, 
```js
const sourceKeys = [
  { 
      id: 1, 
      title: "URL",
      pairPath: APP_CONSTANTS.PATH_FROM_PAIR.RULE_KEYS,
      ruleKey: GLOBAL_CONSTANTS.RULE_KEYS.URL
  },
      .....
]

sourceKeys.map(({id, title, pairPath, ruleKey}) => (
  <Menu.Item key={id} onClick={(e) => modifyPairAtGivenPath(e,pairIndex, pairPath, ruleKey)}
   >{title}</Menu.Item>
))


```

File: app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/RequestSourceRow/index.js [hunk 2]
```diff
@@ -26,108 +26,101 @@ const RequestSourceRow = ({
...
+        key={1}
+        onClick={(event) =>
+          modifyPairAtGivenPath(
+            event,
+            pairIndex,
+            APP_CONSTANTS.PATH_FROM_PAIR.RULE_OPERATORS,
+            GLOBAL_CONSTANTS.RULE_OPERATORS.EQUALS
+          )
+        }
+      >
```
Comment (rohanmathur91) [inline app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/RequestSourceRow/index.js:83]: similarly for source operator options

**Review by rohanmathur91 — CHANGES_REQUESTED** [review]

**Review by rohanmathur91 — APPROVED** [review]

File: app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/RequestSourceRow/index.js [hunk 3]
```diff
@@ -142,7 +119,7 @@ const RequestSourceRow = ({
       wrap={false}
     >
       <Col className="shrink-0">
-        <RQDropdown overlay={sourceKeyOptions} disabled={isInputDisabled}>
+        <RQDropdown overlay={renderSourceKeys()} disabled={isInputDisabled}>
```
Comment (lazyvab) [inline app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/RequestSourceRow/index.js:122]: Calling the function in each render cycle is not optimized.
Rather, you should use `useMemo`.
```js
const sourceKeysMenu = useMemo(() => (
  <Menu>
  </Menu>
), [sourceKeys]);
```

File: app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/RequestSourceRow/index.js [hunk 4]
```diff
@@ -153,7 +130,10 @@ const RequestSourceRow = ({
         </RQDropdown>
       </Col>
       <Col className="shrink-0">
-        <RQDropdown overlay={sourceOperatorOptions} disabled={isInputDisabled}>
+        <RQDropdown
+          overlay={renderSourceOperators()}
```
Comment (lazyvab) [inline app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/RequestSourceRow/index.js:134]: same as above.

File: app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/RequestSourceRow/index.js [hunk 5]
```diff
@@ -107,7 +107,7 @@ const RequestSourceRow = ({
         ))}
       </Menu>
     );
-  };
+  }, [sourceOperators]);
```
Comment (lazyvab) [inline app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/RequestSourceRow/index.js:110]: Shouldn't `modifyPairAtGivenPath` be added to dependencies?

**Review by lazyvab — APPROVED** [review]

**Pull request merged**